### PR TITLE
65 app movies page

### PR DESCRIPTION
**Pull request by Julien-Goletto**
Gros gros morceau lezzzzzzzzgoooo

### CODE REVIEW

File: app/components/Input/Filter.tsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    { title: 'Année de sortie', stateName: 'releaseYear' },
+    { title: 'Note moyenne', stateName: 'avgRate' },
+    { title: 'Pays de production', stateName: 'countries' },
+  ];
+  if (isUserConnected) categories.splice(0, 0, { title: 'Mes actions', stateName: 'review' });
+
+  const reviewActions = [
+    { title: 'Ma liste', stateName: 'bookmarked', svg: <BookmarkSvg style="absolute right-7 w-5 h-5 stroke-white fill-none"/> },
+    { title: 'Non-vus', stateName: 'unwatched', svg: <UnwatchedSvg style="absolute right-[26px] -top-0.5 w-6 h-6 "/> },
+    { title: 'Mes Favoris', stateName: 'liked', svg: <LikeSvg style="absolute right-7 w-5 h-5 stroke-white fill-none"/> },
```
Comment (BenoitSafari) [inline app/components/Input/Filter.tsx:53]: I would do something like 
```typescript
enum ESVGNames {
  BOOKMARKED = "Ma liste",
  UNWATCHED = "Non-vus",
  LIKED = "Mes Favoris"
}
```

File: app/components/Input/Interactions.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import type { FormEvent } from 'react';
 import { useRef } from 'react';
-import { BookmarkSvg, LikeSvg, StarSvg, ViewSvg } from './SVG';
+import { BookmarkSvg, LikeSvg, StarSvg, ViewSvg } from '@components/SvgComponents/InteractionsSVG';
```
Comment (BenoitSafari) [inline app/components/Input/Interactions.tsx:3]: Isn't that more coplicated path than `'./SVG'`? You should add an alias to SvgComponents folder

File: app/components/Input/SearchBar.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import Select from './Select';
+import type { SelectProps } from './Select';
+import type { ControlledTextProps } from './Text';
+
+interface SearchBarProps extends SelectProps, ControlledTextProps<HTMLInputElement> {};
+
+export default function SearchBar(props: SearchBarProps) {
```
Comment (BenoitSafari) [inline app/components/Input/SearchBar.tsx:8]: Refer to discord design discussion

File: app/custom_types/filmsPage.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  filtersCounter: string[];
+  genres?: string[];
+  countries?: string[];
+  runtime?: string[];
+  releaseYear?: string[];
+  avgRate?: string[];
+  [key: string]: string[] | undefined;
+}
+
+export interface filteredMoviesStateInterface {
```
Comment (BenoitSafari) [inline app/custom_types/filmsPage.ts:26]: If you start your other interfaces with a caps you should do it here too

File: app/custom_types/movies.ts [hunk 5]
```diff
@@ -58,4 +58,5 @@ export interface CompleteMovie extends MinimalMovie {
   user_review?: UserReview;
   created_at: string;
   updated_at: string | null;
+  [key: string]: string[] | number | string | boolean | Presentation | Metrics | Comment[] | UserReview | null | undefined
```
Comment (BenoitSafari) [inline app/custom_types/movies.ts:61]: I call that `any`

File: app/hooks/useCloseMenuOnOutsideClick.tsx [hunk 6]
```diff
@@ -2,13 +2,13 @@ import { useEffect } from 'react';
...
 /**
  * @param ref Menu component to close
- * @param name additionnal class name to identify a triggering button user | modal | burger
+ * @param name additionnal class name to identify a triggering button
  * @param state boolean - containing opened status
  * @param setter status mutator
  */
 export default function useCloseMenuOnOutsideClick(
   ref: React.RefObject<HTMLElement>,
-  name: 'user' | 'modal' | 'burger' | 'rating',
```
Comment (BenoitSafari) [inline app/hooks/useCloseMenuOnOutsideClick.tsx:11]: I would use an enum

**Review by BenoitSafari — APPROVED** [review]

File: app/components/Input/Button.tsx [hunk 7]
```diff
@@ -2,10 +2,11 @@ import Link from 'next/link';
 
 interface ButtonProps {
   children?: React.ReactNode;
-  customStyle?: 'text' | 'empty' | 'rounded' | 'white';
+  customStyle?: 'text' | 'empty' | 'rounded' | 'white' | 'select' | 'selectSearchBar';
```
[the remark this answers is not in the document]
Comment (Julien-Goletto) [reply app/components/Input/Button.tsx:5]: Issue dedicacted #83

**Pull request merged**